mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -191,6 +191,8 @@ public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
 			for ( int i = 0; i < searchObjects.length; i++ ) {
 				if ( dbObject.get( ID_FIELDNAME ).equals( searchObjects[i] ) ) {
 					tuples.set( i, createTuple( keys[i], tupleContext, dbObject ) );
```

Comment:
> that's not premature optimisation, that's just pointless overhead.

Yes, that could go away with the map approach. To be fair it was needed in the current approach, driven by the actual results. But yet another reason to change it...